### PR TITLE
[Feature Request] Adds helm chart for mnist example

### PR DESCRIPTION
**Pull request by techjacker**
---
name: Feature request - Adds helm chart deployment for mnist example
about: Adds the ability to deploy the mnist example to kubernetes via a helm chart
---

## Summary
Adds the ability to deploy the mnist example to kubernetes via a helm chart.
The source code of library had to be altered slighlty to better facilitate  containerisation.

## Motivation
We wanted to deploy the mnist example to the cloud.

## Describe alternatives you've considered
Serverless platforms such as AWS lambda. Kubernetes is the defacto standard in cloud deployment and is platform agnostic so was the obvious choice to start with.

## Additional context
Recommend you package the helm chart and push it to a chart repo of your choice.

### CODE REVIEW

File: src/dc_federated/algorithms/fed_avg/fed_avg_server.py [hunk 1]
```diff
@@ -58,15 +57,7 @@ class FedAvgServer(object):
...
-    def __init__(self,
-                 global_model_trainer,
-                 key_list_file,
-                 update_lim=10,
-                 server_host_ip=None,
-                 server_port=8080,
-                 ssl_enabled=False,
-                 ssl_keyfile=None,
-                 ssl_certfile=None):
+    def __init__(self, server_port, global_model_trainer, key_list_file, update_lim=10, server_host_ip=None, ssl_enabled=False, ssl_keyfile=None, ssl_certfile=None):
```
Comment (hassan-digicatapult) [inline src/dc_federated/algorithms/fed_avg/fed_avg_server.py:60]: @techjacker Why did you move this parameter forward?

Comment (techjacker) [reply src/dc_federated/algorithms/fed_avg/fed_avg_server.py:60]: My thinking was to make it a required rather than optional parameter as it will be fed in via the helm chart and to avoid debugging issues arising from a default port being set.
